### PR TITLE
Improve visualizer aesthetics

### PR DESCRIPTION
**Pull request by pculbertson**
A medium-sized feature allowing us to apply MuJoCo textures to meshes. This was pretty rushed and likely needs more attention - the test coverage especially is not good, and I think there will be plenty of edge cases we should sort through more carefully (e.g., what happens if both the mesh has textures defined in mujoco and in the `.obj`?).

Still, the Franka does look much nicer now. 

Main changes: 
* Swapped the base `Task` to carry around an `MjSpec` (which has the asset filepaths inside it).
* Swapped the `ViserMjModel` to use this `MjSpec` to locate mesh assets and textures that will be better visualized in `viser`.
* Updated geom/body naming inside the `ViserMjModel`.
* Updated `ViserMjModel` unit tests to comply with the new interface.

### CODE REVIEW

**Review by alberthli — APPROVED** [review]
lgtm

File: judo/visualizers/utils.py [hunk 1]
```diff
@@ -38,6 +43,103 @@ def get_mesh_data(model: MjModel, meshid: int) -> tuple[np.ndarray, np.ndarray]:
...
+    # Fallback for legacy shininess value if roughness is not set
+    if material.roughness == 0.0 and hasattr(material, "shininess") and material.shininess > 0:
+        pbr_material.roughnessFactor = np.sqrt(2.0 / (material.shininess + 2.0))
+
+    # --- 2. Handle Textured vs. Simple Color cases ---
+    has_texture = hasattr(material, "textures") and material.textures and material.textures[0]
+    texture_loaded = False
+
+    if has_texture:
+        texture_path = texture_dir / material.textures[0]
```
[the remark this answers is not in the document]
Comment (pculbertson) [reply judo/visualizers/utils.py:116]: No but then you hit a special case where you need like a `MultiMaterial` or something (forget the exact name) from `trimesh`, and need to bookkeep which faces get what texture. We could maybe throw a more informative error in the multi-texture case (so your QWERTY cube will fail loudly)

File: judo/visualizers/utils.py [hunk 2]
```diff
@@ -38,6 +43,103 @@ def get_mesh_data(model: MjModel, meshid: int) -> tuple[np.ndarray, np.ndarray]:
...
+    pbr_material.baseColorFactor = rgba
+    pbr_material.alphaMode = "BLEND" if rgba[-1] < 255 else "OPAQUE"
+
+    # Map PBR properties
+    pbr_material.metallicFactor = material.metallic
+    pbr_material.roughnessFactor = material.roughness
+    pbr_material.emissiveFactor = np.array([material.emission] * 3)
+
+    # Fallback for legacy shininess value if roughness is not set
+    if material.roughness == 0.0 and hasattr(material, "shininess") and material.shininess > 0:
```
[the remark this answers is not in the document]
Comment (slecleach) [reply judo/visualizers/utils.py:108]: Yes that makes sense, we take another pass after 0.0.2 to robustify if needed.

**Review by slecleach — APPROVED** [review]
